### PR TITLE
Feature/inference

### PR DESCRIPTION
**Pull request by jaxony**
@gngdb I've added some inference code to infer on an cat image using the shufflenet checkpoint that you provided. Could you please checkout this branch and see if I've done anything stupid? It doesn't predict the categories correctly at the moment (maybe it's just a hard image, but I doubt that!)

### CODE REVIEW

File: infer.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import numpy as np
+
+def get_transformer():
+  normalize = transforms.Normalize(
+      mean=[0.485, 0.456, 0.406],
+      std=[0.229, 0.224, 0.225])
+
+  transformer = transforms.Compose([
+      transforms.Resize(128),
+      transforms.CenterCrop(128),
```
Comment (gngdb) [inline infer.py]: No point in cropping after resize, you can remove this line.

File: infer.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+def preprocess(image, transformer):
+  x = transformer(image)
+  return Variable(x.unsqueeze(0))
+
+def infer(args):
+  # make ShuffleNet model
+  print('Creating ShuffleNet model')
+  net = ShuffleNet(num_classes=args.num_classes, in_channels=3)
+
```
Comment (gngdb) [inline infer.py:33]: The network should be in `.eval()` mode during inference; add `net.eval()` here.

**Review by gngdb — CHANGES_REQUESTED** [review]
After adding the `net.eval()` line and it worked. Must have been a batchnorm problem at inference time.

File: infer.py (hunk 2 above)
Comment (jaxony) [reply infer.py:33]: Ah that's right. I've made this mistake at least three times--and still don't remember to put it in.

**Pull request merged**